### PR TITLE
Documented and exposed the control group sprites to lint testing

### PR DESCRIPTION
**Pull request by Mailaender**
Not setting them up will cause problems to modders.

### CODE REVIEW

**Comment by abcdefg30** [comment]
👍

File: OpenRA.Mods.Common/Traits/Render/SelectionDecorations.cs [hunk 1]
```diff
@@ -104,9 +109,9 @@ IEnumerable<IRenderable> DrawControlGroup(WorldRenderer wr, Actor self, int2 bas
 			if (group == null)
 				yield break;
 
-			var pipImages = new Animation(self.World, "pips");
+			var pipImages = new Animation(self.World, Info.Image);
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Traits/Render/SelectionDecorations.cs]: This should be cached in the constructor.

File: OpenRA.Mods.Common/Traits/Render/SelectionDecorations.cs [hunk 2]
```diff
@@ -104,9 +109,9 @@ IEnumerable<IRenderable> DrawControlGroup(WorldRenderer wr, Actor self, int2 bas
...
 			var pal = wr.Palette(Info.Palette);
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Traits/Render/SelectionDecorations.cs]: Same here, while you're at it.

File: OpenRA.Mods.Common/Traits/Render/SelectionDecorations.cs [hunk 3]
```diff
@@ -119,7 +124,7 @@ IEnumerable<IRenderable> DrawPips(WorldRenderer wr, Actor self, int2 basePositio
 			if (!pipSources.Any())
 				yield break;
 
-			var pipImages = new Animation(self.World, "pips");
+			var pipImages = new Animation(self.World, Info.Image);
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Traits/Render/SelectionDecorations.cs]: Same here.

Comment (reaperrr) [inline OpenRA.Mods.Common/Traits/Render/SelectionDecorations.cs]: and same for [line 133](https://github.com/Mailaender/OpenRA/blob/3a7558c273bdebd2e7466173331f1100d9ece1e8/OpenRA.Mods.Common/Traits/Render/SelectionDecorations.cs#L133)

**Comment by reaperrr** [comment]
I think this might be a good opportunity to unhardcode [PipStrings](https://github.com/Mailaender/OpenRA/blob/3a7558c273bdebd2e7466173331f1100d9ece1e8/OpenRA.Mods.Common/Traits/Render/SelectionDecorations.cs#L50) as `PipSequences` with the comment added to the property description.

File: OpenRA.Mods.Common/Traits/Render/SelectionDecorations.cs (hunk 2 above)
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Render/SelectionDecorations.cs]: I don't have access to `WorldRenderer` in the constructor.

**Comment by pchote** [comment]
In the long term we should move the pip definitions out of this trait completely (to the traits that want to show the pips).

**Comment by reaperrr** [comment]
> I will stay clear of PipStrings for now as it will require all pips to be setup.

Fair enough.

**Pull request merged**